lib/task_listener.js

```diff
@@ -110,21 +108,23 @@ export default class TaskListener extends EventEmitter {
     return cancelListener;
   }
 
+  get availableCapacity() {
+    return Math.max(this.capacity - this.runningTasks.length, 0);
```

Comment:
Maybe log an `[alert-operator]` if `this.capacity - this.runningTasks.length` is negative... I mean that should never be able to happen, right?